book/docs/verification/onchain/solidity-sdk.md

```diff
@@ -64,6 +64,14 @@ contract Fibonacci {
 
 ```
 
+The recommended workflow is to use the `ISP1Verifier` interface on the [`SP1VerifierGateway`](./contract-addresses.md),
+and the `SP1VerifierGateway` will automatically route your proof to the correct verifier.
+
+It’s also recommended to have your program’s verification key be upgradeable within your contract, in case you want to switch to a different version of SP1.
```

Comment:
```suggestion
Your program’s verification key should be upgradeable within your contract, in case you want to upgrade to a newer version of SP1 or modify your program.
```